web_interface/blueprints/api_v3.py

```
@@ -1341,6 +1341,57 @@ def get_system_version():
+@api_v3.route('/system/check-update', methods=['GET'])
+def check_for_update():
+    """Check if a newer version is available on the remote."""
+    import time as _time
+    now = _time.time()
+    if _update_check_cache.get('ts', 0) + _UPDATE_CHECK_TTL > now:
+        return jsonify(_update_check_cache['data'])
+
+    project_dir = str(PROJECT_ROOT)
+    try:
+        subprocess.run(
+            ['git', 'fetch', 'origin', 'main'],
```

Ruff (0.15.10)

[error] Starting a process with a partial executable path

(S607)